client/network/src/service.rs

```diff
@@ -1045,21 +1052,35 @@ impl<B: BlockT + 'static, H: ExHashT> Future for NetworkWorker<B, H> {
 						error,
 					);
 
-					if let Some(peer_id) = peer_id {
-						if this.boot_node_ids.contains(&peer_id)
-							&& error.contains("Peer ID mismatch")
-						{
-							error!(
-								"Connecting to bootnode with peer id `{}` and address `{}` failed \
-								because it returned a different peer id!",
-								peer_id,
-								address,
-							);
-						}
+					if this.boot_node_ids.contains(&peer_id)
+						&& error.contains("Peer ID mismatch")
```

Comment:
This is a hack because the error used to be a `Box<dyn Error>`, while in libp2p 0.17 it is now an enum (#5369).
Feel free to fix this now, or someone can fix later separately.

Reply:
I can do that here, this was just a blind code adaptation as part of a rebase.